android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -140,94 +140,99 @@ constructor(
 
   /**
    * This function retrieves the [Questionnaire] as configured via the [QuestionnaireConfig]. The
-   * retrieved [Questionnaire] can be pre-populated with computed values from the Rules engine as
-   * well as include initial values set on configured [QuestionnaireConfig.barcodeLinkId] or
-   * [QuestionnaireConfig.uniqueIdAssignment] properties.
+   * retrieved [Questionnaire] can then be pre-populated.
    */
   suspend fun retrieveQuestionnaire(
     questionnaireConfig: QuestionnaireConfig,
-    actionParameters: List<ActionParameter>?,
   ): Questionnaire? {
     if (questionnaireConfig.id.isEmpty() || questionnaireConfig.id.isBlank()) return null
+    return defaultRepository.loadResource<Questionnaire>(questionnaireConfig.id)
+  }
+
+  /**
+   * Pre-populates questionnaire with computed values from the Rules engine as well as include
+   * initial values set on configured [QuestionnaireConfig.barcodeLinkId] or
+   * [QuestionnaireConfig.uniqueIdAssignment] properties.
+   */
+  private suspend fun Questionnaire.prepopulateWithComputeConfigValues(
+    config: QuestionnaireConfig,
+    actionParameters: List<ActionParameter>?,
+  ) {
+    require(config.id.isNotBlank())
 
     // Compute questionnaire config rules and add extra questionnaire params to action parameters
     val questionnaireComputedValues =
-      questionnaireConfig.configRules?.let {
+      config.configRules?.let {
```

Comment:
Is there a reason we no longer want the config variable explicitly named as it was before?